acoular/spectra.py

```diff
@@ -294,8 +300,17 @@ class PowerSpectra(BaseSpectra):
 
     @property_depends_on('_source.numsamples, block_size, overlap')
     def _get_num_blocks(self):
+        if self._num_blocks != 0: # The source is a real time stream
+            return self._num_blocks 
         return self.overlap_ * self._source.numsamples / self.block_size - self.overlap_ + 1
 
+    def _set_num_blocks(self, value):
```

Comment:
This should simply set the shadow trait, because setting num_blocks could also be of use for non streaming data. However, it should never be larger than `self.overlap_ * self._source.numsamples / self.block_size - self.overlap_ + 1`.

Reply:
Added a check for this condition https://github.com/acoular/acoular/pull/285/commits/e156e01325b09ece2aa8ac2ba9572accd96e24af